### PR TITLE
makefile: Add a way to add replaces,skips in the CSV

### PR DESCRIPTION
**Pull request by iamniting**
REPLACES, SKIP_RANGE is required by OLM while upgrading a CSV to the newer version.

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=2126708

### CODE REVIEW

**Review by leelavg — APPROVED** [review]
- just confirming, this is required for upgrade of operator via OLM right?

**Comment by iamniting** [comment]
> * just confirming, this is required for upgrade of operator via OLM right?

yes right

File: config/manifests/bases/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- lvm-operator.clusterserviceversion.yaml
```
Comment (nbalacha) [inline config/manifests/bases/kustomization.yaml:4]: The name of this file can change (downstream build uses odf-lvm-operator.clusterserviceversion.yaml) depending on the value of $BUNDLE_PACKAGE (see rename-csv)

File: Makefile [hunk 2]
```diff
@@ -221,6 +231,9 @@ bundle: update-mgr-env manifests kustomize operator-sdk rename-csv build-prometh
 	cd config/default && $(KUSTOMIZE) edit set namespace $(OPERATOR_NAMESPACE)
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	cd config/default && $(KUSTOMIZE) edit set image rbac-proxy=$(RBAC_PROXY_IMG)
+	cd config/manifests/bases && $(KUSTOMIZE) edit add annotation --force 'olm.skipRange':"$(SKIP_RANGE)" && \
+		$(KUSTOMIZE) edit add patch --name lvm-operator.v0.0.0 --kind ClusterServiceVersion\
```
Comment (nbalacha) [inline Makefile:235]: The operator name also changes based on whether this is upstream or downstream.

File: config/manifests/bases/kustomization.yaml (hunk 1 above)
Comment (iamniting) [reply config/manifests/bases/kustomization.yaml]: Thanks for pointing it out, made changes accordingly.

File: Makefile (hunk 2 above)
Comment (iamniting) [reply Makefile]: same here

File: Makefile [hunk 3]
```diff
@@ -221,6 +231,12 @@ bundle: update-mgr-env manifests kustomize operator-sdk rename-csv build-prometh
 	cd config/default && $(KUSTOMIZE) edit set namespace $(OPERATOR_NAMESPACE)
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	cd config/default && $(KUSTOMIZE) edit set image rbac-proxy=$(RBAC_PROXY_IMG)
+	cd config/manifests/bases && \
+		rm -rf kustomization.yaml && \
+		$(KUSTOMIZE) create --resources $(BUNDLE_PACKAGE).clusterserviceversion.yaml && \
+		$(KUSTOMIZE) edit add annotation --force 'olm.skipRange':"$(SKIP_RANGE)" && \
+		$(KUSTOMIZE) edit add patch --name $(BUNDLE_PACKAGE).v0.0.0 --kind ClusterServiceVersion\
```
Comment (leelavg) [inline Makefile:238]: - i think it's better to have `$(BUNDLE_PACKAGE).v$(VERSION)` rather than `$(BUNDLE_PACKAGE).v0.0.0`

Comment (iamniting) [reply Makefile:238]: This version is coming from this file [1] which has to be exactly the same always.

[1]. https://github.com/red-hat-storage/lvm-operator/blob/main/config/manifests/bases/clusterserviceversion.yaml.in#L7

**Comment by leelavg** [comment]
@iamniting anything pending on this PR from your side?

**Comment by iamniting** [comment]
> @iamniting anything pending on this PR from your side?

nothing is pending.

**Review by leelavg — APPROVED** [review]
/lgtm

**Comment by iamniting** [comment]
lvm-operator has been transferred to a different team. They will create PR according to their own cadence.

**Pull request closed without merging**